Bring back sticky table header on safari

stipsan opened the pull request:
Firefox still don't do sticky table headers very well, but Safari do.

It makes it so much easier to read the table as the headers are always visible.

What it looks like on Safari 8.0.5 on Yosemite:

![animation](https://cloud.githubusercontent.com/assets/81981/7451544/2b565634-f252-11e4-8194-0f72971cf22a.gif)

stipsan committed:
Bring back sticky table header on safari

Firefox still don't do sticky table headers very well, but Safari do.

It makes it so much easier to read the table as the headers are always visible.

kangax commented:
@stipsan thanks!
@webbedspace i recall something about us trying this before but then reverting back? Do you remember by any chance? Was it buggy implementation or smth else?

webbedspace commented:
`position:sticky` or its vendor-prefix versions don't work for `<tr>` elements in Firefox, hence I withdrew it on that basis.

I'm a bit leery of accepting this as-is because I don't like the look of the transparent cell borders - aesthetically, they should really be solid white for the duration of it being stuck. This is a bit difficult to do for a `<tr>` in CSS, admittedly.

kangax commented:
I agree about solid white. As for partial support — I'm ok with it. I know
that Chrome is working on it (
https://code.google.com/p/chromium/issues/detail?id=231752) and here's
Mozilla's ticket — https://bugzilla.mozilla.org/show_bug.cgi?id=975644 so
I'm guessing we'll have much wider support soon.

On Mon, May 4, 2015 at 4:01 PM, Leon notifications@github.com wrote:

> position:sticky or its vendor-prefix versions don't work for <tr>
> elements in Firefox, hence I withdrew it on that basis.
> 
> I'm a bit leery of accepting this as-is because I don't like the look of
> the transparent cell borders - aesthetically, they should really be solid
> white for the duration of it being stuck. This is a bit difficult to do for
> a <tr> in CSS, admittedly.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kangax/compat-table/pull/532#issuecomment-98715499.

claudepache commented:
> I'm a bit leery of accepting this as-is because I don't like the look of the transparent cell borders - aesthetically, they should really be solid white for the duration of it being stuck.

For simulating solid white borders, one may hack with the outline of cells:

``` js
thead > tr:nth-child(2) > th {
    outline: 1px solid white;
}
```

There is still a problem with column headers of unstable and obsolete platforms, as their background is semi-transparent:

![c](https://cloud.githubusercontent.com/assets/3309637/8992484/00a69c90-36ff-11e5-85ec-23482e5114e1.png)

stipsan committed:
Fixed transparent cell borders, added prefixless

Firefox is landing table support for position fixed soon and MS is considering including position sticky for Edge.

stipsan commented:
@claudepache I adjusted it to 2px so the bottom outline match the overall cell spacing.

The semi-transparent background is fixed.

@webbedspace here's what it looks like now:
![compat](https://cloud.githubusercontent.com/assets/81981/9303074/06d16b82-44dc-11e5-9b6b-8b4722638f84.gif)

webbedspace commented:
![](http://i.imgur.com/E0a361L.png)
The top header looks like this in Firefox, which is somewhat unpleasant.

stipsan committed:
Fix weird border regression in firefox

stipsan commented:
@webbedspace indeed, using box-shadow instead of outline solves it in FF while looking exactly the same in Safari, Chrome etc.

There's a plugin for cross browser compat: https://github.com/filamentgroup/fixed-sticky
But it's not yet fully working with table thead elements yet.

MS is considering pos sticky for Edge, and FF is working on a fix for sticky table elements thus I vendor prefixed those too.

stipsan commented:
@webbedspace what do you think of it now?
